spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -868,13 +875,16 @@ else if (FileExistsMode.IGNORE != fileExistsMode) {
 			return mGetWithRecursion(message, session, remoteDirectory, remoteFilename);
 		}
 		else {
+			remoteDirectory = (remoteDirectory != null
+					? remoteDirectory
+					: "." + this.remoteFileTemplate.getRemoteFileSeparator());
```

Comment:
No; we need to support MGET with a null path; especially if no recursion - but we should allow, even with recursion.